libindy-crypto/src/ffi/cl/prover.rs

```diff
@@ -0,0 +1,633 @@
+                                                            issuer_pub_key: *const c_void,
+                                                            rev_reg_pub: *const c_void) -> ErrorCode {
+    trace!("indy_crypto_cl_prover_process_claim_signature: >>> claim_signature: {:?}, master_secret_blinding_data: {:?}, issuer_pub_key: {:?}, rev_reg_pub: {:?}",
+           claim_signature, master_secret_blinding_data, issuer_pub_key, rev_reg_pub);
+
+    check_useful_mut_c_reference!(claim_signature, ClaimSignature, ErrorCode::CommonInvalidParam1);
+    check_useful_c_reference!(master_secret_blinding_data, MasterSecretBlindingData, ErrorCode::CommonInvalidParam2);
+    check_useful_c_reference!(issuer_pub_key, IssuerPublicKey, ErrorCode::CommonInvalidParam3);
+    check_useful_opt_c_reference!(rev_reg_pub, RevocationRegistryPublic, ErrorCode::CommonInvalidParam4);
+
+    trace!("indy_crypto_cl_prover_process_claim_signature: entities: claim_signature: {:?}, master_secret_blinding_data: {:?}, issuer_pub_key: {:?}, rev_reg_pub: {:?}",
+           claim_signature, master_secret_blinding_data, issuer_pub_key, rev_reg_pub);
+
+    let res = match Prover::process_claim_signature(claim_signature,
+                                                    master_secret_blinding_data,
+                                                    issuer_pub_key,
+                                                    rev_reg_pub) {
+        Ok(()) => ErrorCode::Success,
+        Err(err) => err.to_error_code()
+    };
+
+    trace!("indy_crypto_cl_prover_process_claim_signature: <<< res: {:?}", res);
+    res
+}
+
+/// Creates and returns proof builder.
+///
+/// The purpose of proof builder is building of proof entity according to the given request .
+///
+/// Note that proof builder deallocation must be performed by
+/// calling indy_crypto_cl_proof_builder_finalize
+///
+/// # Arguments
+/// * `proof_builder_p` - Reference that will contain proof builder instance pointer.
+#[no_mangle]
+pub extern fn indy_crypto_cl_prover_new_proof_builder(proof_builder_p: *mut *const c_void) -> ErrorCode {
+    trace!("indy_crypto_cl_prover_new_proof_builder: >>> {:?}", proof_builder_p);
+
+    check_useful_c_ptr!(proof_builder_p, ErrorCode::CommonInvalidParam1);
+
+    let res = match Prover::new_proof_builder() {
+        Ok(proof_builder) => {
+            trace!("indy_crypto_cl_prover_new_proof_builder: proof_builder: {:?}", proof_builder);
+            unsafe {
+                *proof_builder_p = Box::into_raw(Box::new(proof_builder)) as *const c_void;
+                trace!("indy_crypto_cl_prover_new_proof_builder: *proof_builder_p: {:?}", *proof_builder_p);
+            }
+            ErrorCode::Success
+        }
+        Err(err) => err.to_error_code()
+    };
+
+    trace!("indy_crypto_cl_prover_new_proof_builder: <<< res: {:?}", res);
+    res
+}
+
+/// Add sub proof request to proof builder which will be used fo building of proof.
```

Comment:
Please describe what sub-proof request is (this is part of proof request related to a particular schema-key)